### DISCUSSION TITLE
Using redirects/rewrites when [...slug].js is in the pages root

### DISCUSSION THREAD

**Discussion by mylesthedev** [vercel/next.js#19671] [Help]
It seems that when a page like `[slug].js` or `[....slug].js` is in the folder pages (e.g. not in a subfolder) that redirects or rewrites don't work due to the 404 being triggered first at build time.

Most of the sites I build use a CMS so users can create pages with any URL hence the need for `[...slug].js`.

Does anyone have a workaround for this? Or can suggest a better way to handle this scenario?

_[EDIT] By the way I've tried both vercal.json rewrites and next.config.js rewrites and redirects_

**Comment by mylesthedev** [comment]
A possible workaround I thought of was using the router inside the 404 function to detect the requested page.

e.g. 
```js
function Custom404() {
    let router = useRouter();
    useEffect(() => {
        if(router.asPath == '/success-stories') {
            router.push('/work')
        }
    })
}
```

**Comment by leerob** [comment]
Make sure you have `fallback: false` so that it doesn't try to query for pages that aren't available. That conflicts with the rewrites because it's trying to query for a non-existing page (if you have a fallback) when it should be rewriting to the proxy URL. 

If you add `fallback: false`, then if the route doesn't exist, it will rewrite to your proxy URL.

**Reply by mylesthedev** [reply]
Thanks @leerob but this particular page was already set to `fallback: false`.
I've also tried it with and without a custom 404 template just incase it was that.

**Comment by mylesthedev** [comment]
Also found this in the rewrites example repo...
```
// Use SSR for this page as currently rewrites don't work with dynamic pages without SSR
```

**Comment by bscaspar** [comment]
Hey @mediabeastnz did you ever figure out a solution to this? I'm running into the same thing with a `/[slug]` page and a rewrite at `/[slug]/[id]` getting a 404, even with `fallback: false` and with my `rewrites` config using the `fallback` object.
